Index: app/src/main/java/com/cyruszhang/cluboard/ClubDetail.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/cyruszhang/cluboard/ClubDetail.java	(revision 1111a5d2ca3174693f0fac33bd9e6e95074660fb)
+++ app/src/main/java/com/cyruszhang/cluboard/ClubDetail.java	(revision )
@@ -154,10 +154,13 @@
         bookmark.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS);
         bookmark.setCheckable(true);
         // TODO: returns false all the time???
-        if (((User)(User.getCurrentUser())).checkBookmarkClub(thisClub)) {
+        User thisUser = (User) ParseUser.getCurrentUser();
+        if (thisUser.checkBookmarkClub(thisClub)) {
+            Log.d(getClass().getSimpleName(), "You have bookmarked it before.");
             bookmark.setChecked(true);
             bookmark.setIcon(R.drawable.ic_action_remove_bookmark);
         } else {
+            Log.d(getClass().getSimpleName(), "You have not bookmarked it before.");
             bookmark.setChecked(false);
             bookmark.setIcon(R.drawable.ic_action_add_bookmark);
         }
